Mathlib/Tactic/Linter/DeprecateModule.lean

```diff
@@ -0,0 +1,138 @@
+/-
+Copyright (c) 2025 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import Mathlib.Init
+
+/-!
+#  The "deprecateModule" linter
+
+The "deprecateModule" linter emits a warning when a file that has been renamed or split
+is imported.
+
+The usage is as follows. Write
+```lean
+import B
+...
+import Z
+
+deprecate_module since yyyy-mm-dd
+```
+in module `A` with the expectation that `A` contains nothing else.
+This triggers the `deprecateModule` linter to notify every file with `import A`
+to instead import the *direct imports* of `A`, that is `B,...,Z`.
+-/
+
+open Lean Elab Command
+
+namespace Mathlib.Linter
+
+/--
+The `deprecateModule` linter emits a warning when a file that has been renamed or split
+is imported.
```

Comment:
Do you think it's useful to mention that this is enabled in dependencies of mathlib on purpose (so they will get import deprecation warnings)?

Reply:
Added a sentence!